pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -467,12 +468,35 @@ func resolveConditionChecks(pt *v1alpha1.PipelineTask,
 			}
 		}
 
-		rcc = append(rcc, &ResolvedConditionCheck{
+		rcc := ResolvedConditionCheck{
 			Condition:             c,
 			ConditionCheckName:    conditionCheckName,
 			ConditionCheck:        v1alpha1.NewConditionCheck(cctr),
 			PipelineTaskCondition: &ptc,
-		})
+		}
+
+		if len(ptc.Resources) > 0 {
+			rr := make(map[string]*v1alpha1.PipelineResource)
+			for _, r := range ptc.Resources {
+				// First get a ref to actual resource name from its bound name
+				resourceRef, ok := providedResources[r.Resource]
+				if !ok {
+					return nil, xerrors.Errorf("condition %s for pipeline task %s tried to use resource %s not present in declared resources", cName, pt.Name, r.Resource)
+				}
+
+				// Next, fetch the actual resource definition
+				gotResource, err := getResource(resourceRef.Name)
+				if err != nil {
+					return nil, xerrors.Errorf("could not retrieve resource %s referenced in condition %s for pipeline task %s", r.Name, cName, pt.Name)
+				}
+
+				// Finally add it to the resolved resources map
+				rr[r.Name] = gotResource
+			}
+			rcc.ResolvedResources = rr
```

Comment:
this kinda feels like it could be a separate function, what do you think? 

`resolveConditionChecks` could even become `ResolveConditionChecks` and be called by the reconciler after calling `ResolvePipelineRun` - that would make it so `ResolvePipelineRun` and its unit tests could be simpler, and you'd no longer need to distinguish between different kinds of errors

Reply:
ok, I split the resource resolution into a separate function. There is definitely a lot of scope to refactor the pipelineState resolution -- besides resolving the conditionChecks separately, another thing that might help (particularly with #1023) would be changing the `PipelineState` from just being a slice of the ResolvedTasks to  having a reference to the `Dag` nodes.

Reply:
> There is definitely a lot of scope to refactor the pipelineState resolution

Yeah probably too much to do all at ounce - but thanks for the incremental improvement!